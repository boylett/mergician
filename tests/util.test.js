const {
    countOccurrences,
    getInMultiple,
    getInAll,
    getNotInMultiple,
    getNotInAll,
    isObject
} = require('../src/util');

const arr1 = ['a', 'b', 'c'];
const arr2 = ['b', 'c', 'd'];
const arr3 = ['c', 'd', 'e'];

describe('countOccurrences', () => {
    test('generates map', () => {
        const result = countOccurrences(arr1, arr2, arr3);

        expect(result).toMatchSnapshot();
    });
});

describe('getInMultiple', () => {
    test('gets common values in two arrays', () => {
        const result = getInMultiple(arr1, arr2);

        expect(result).toMatchSnapshot();
    });

    test('gets common values found in multiple arrays', () => {
        const result = getInMultiple(arr1, arr2, arr3);

        expect(result).toMatchSnapshot();
    });
});

describe('getInAll', () => {
    test('gets universal values in two arrays', () => {
        const result = getInAll(arr1, arr2);

        expect(result).toMatchSnapshot();
    });

    test('gets universal values in multiple arrays', () => {
        const result = getInAll(arr1, arr2, arr3);

        expect(result).toMatchSnapshot();
    });
});

describe('getNotInMultiple', () => {
    test('gets values found in one of two arrays', () => {
        const result = getNotInMultiple(arr1, arr2);

        expect(result).toMatchSnapshot();
    });

    test('gets values found in one of multiple arrays', () => {
        const result = getNotInMultiple(arr1, arr2, arr3);

        expect(result).toMatchSnapshot();
    });
});

describe('getNotInAll', () => {
    test('gets values not found in two arrays', () => {
        const result = getNotInAll(arr1, arr2);

        expect(result).toMatchSnapshot();
    });

    test('gets values not found in three arrays', () => {
        const result = getNotInAll(arr1, arr2, arr3);

        expect(result).toMatchSnapshot();
    });
});

describe('isObject', () => {
    test('true of object literal', () => {
        expect(isObject({})).toBe(true);
    });

    test('false for array', () => {
        expect(isObject([])).toBe(false);
    });

    test('false for function', () => {
        expect(isObject(Function.prototype)).toBe(false);
    });
});
