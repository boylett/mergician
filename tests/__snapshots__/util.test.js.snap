// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countOccurrences generates map 1`] = `
Object {
  "a": 1,
  "b": 2,
  "c": 3,
  "d": 2,
  "e": 1,
}
`;

exports[`getInAll gets universal values in multiple arrays 1`] = `
Array [
  "c",
]
`;

exports[`getInAll gets universal values in two arrays 1`] = `
Array [
  "b",
  "c",
]
`;

exports[`getInMultiple gets common values found in multiple arrays 1`] = `
Array [
  "b",
  "c",
  "d",
]
`;

exports[`getInMultiple gets common values in two arrays 1`] = `
Array [
  "b",
  "c",
]
`;

exports[`getNotInAll gets values not found in three arrays 1`] = `
Array [
  "a",
  "b",
  "d",
  "e",
]
`;

exports[`getNotInAll gets values not found in two arrays 1`] = `
Array [
  "a",
  "d",
]
`;

exports[`getNotInMultiple gets values found in one of multiple arrays 1`] = `
Array [
  "a",
  "e",
]
`;

exports[`getNotInMultiple gets values found in one of two arrays 1`] = `
Array [
  "a",
  "d",
]
`;
